edgelet/edgelet-kube/src/registry/pull.rs

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::collections::HashMap;
+use std::convert::TryFrom;
+
+use failure::Fail;
+use futures::future::Either;
+use futures::prelude::*;
+use futures::{future, Future, Stream};
+use hyper::service::Service;
+use hyper::Body;
+
+use edgelet_docker::DockerConfig;
+use kube_client::TokenSource;
+
+use crate::convert::NamedSecret;
+use crate::error::{Error, ErrorKind};
+use crate::registry::ImagePullSecret;
+use crate::KubeModuleRuntime;
+
+pub fn create_image_pull_secrets<T, S>(
+    runtime: &KubeModuleRuntime<T, S>,
+    config: &DockerConfig,
+) -> impl Future<Item = (), Error = Error>
+where
+    T: TokenSource + Send + 'static,
+    S: Service + Send + 'static,
+    S::ReqBody: From<Vec<u8>>,
+    S::ResBody: Stream,
+    Body: From<S::ResBody>,
+    S::Error: Fail,
+    S::Future: Send,
+{
+    let image_pull_secrets = vec![config.auth(), runtime.settings().proxy().auth()]
+        .into_iter()
+        .filter_map(|auth| {
+            auth.and_then(|auth| ImagePullSecret::from_auth(&auth))
+                .map(|secret| (secret.name(), secret))
+        })
+        .collect::<HashMap<_, _>>()
+        .drain()
+        .map(|(_, secret)| secret)
```

Comment:
Possible suggestion: 
```
use itertools::Itertools;

let image_pull_secrets = vec![config.auth(), runtime.settings().proxy().auth()]
        .into_iter()
        .filter_map(|auth| {
            auth.and_then(|auth| ImagePullSecret::from_auth(&auth))
        })
        .unique_by(|secret| secret.name())
        .collect::<Vec<_>>();
```

Reply:
I like this idea, but do we want to add additional crate?

Reply:
It's not a required change, I just like the fact I found a more terse way to do the same thing. :)